src/rebuild/srv.c

```diff
@@ -2506,14 +2514,16 @@ rebuild_tgt_prepare(crt_rpc_t *rpc, struct rebuild_tgt_pool_tracker **p_rpt)
 	rpt->rt_pool = pool; /* pin it */
 	ABT_mutex_unlock(rpt->rt_lock);
 
-	rpt_get(rpt);
-	d_list_add(&rpt->rt_list, &rebuild_gst.rg_tgt_tracker_list);
 	*p_rpt = rpt;
 out:
 	if (rc) {
-		if (rpt)
+		if (rpt) {
 			rpt_put(rpt);
-
+			if (!d_list_empty(&rpt->rt_list)) {
+				d_list_del_init(&rpt->rt_list);
+				rpt_put(rpt);
```

Comment:
rpt_put() doesn't destroy 'rpt' when last reference dropped? I'm just wondering who is responsible for removing the 'rpt' from the global list and destroy it? It looks to me that there is no 'definite last reference holder' for 'rpt', right?